docs/contribute/dev-setup.md

```diff
@@ -0,0 +1,112 @@
+
+If the file affects the `preview` (the inner iframe of Storybook, where the stories contents live), the preview should automatically refresh a second or two after you save your changes.
+
+If the file affects the `manager` (the outer iframe where the UI and addon panels live), you need to manually refresh the browser.
+
+## Reproductions
+
+Since May 2021 we're strongly encouraging users to create reproductions for their issues. Just like it's possible to [interactively develop](#interactive-development) against example projects in the storybook monorepo, it's also possible to interactively develop against a reproduction repo.
+
+To do so, run the following command in the root of the Storybook monorepo:
+
+```sh
+npx sb@next link https://github.com/<some-user>/<some-project>.git
+```
+
+This will create a project `../storybook-repros/some-project`, and then automatically link it to your local Storybook code.
+
+After the repro is linked, you should be able to run Storybook and develop interactively with the `yarn build` command just like in the previous section.
+
+```sh
+cd ../storybook-repros/some-project
+yarn storybook
+```
+
+## Check your work
+
+When you're done with your work, make sure you've added documentation and tests as appropriate. If you don't, we'll remind you in your PR. 😘
+
+Adding a story or set of stories to an example app is a great way to test your work. If you're modifying part of Storybook's core, or one of the addons, there's probably a set of stories you can add to in `examples/official-storybook`. If you're modifying something related to a specific framework, they have their own examples in the monorepo. For instance, `examples/vue-kitchen-sink` is a natural place to add stories for `@storybook/vue`, `examples/angular-cli` for `@storybook/angular`, and so on.
+
+Unit tests are also part of our test strategy. The naming convention for unit tests is as follows:
+
+```sh
+# Proper naming convention and structure for ts tests files
++-- parentFolder
+|   +-- [filename].ts
+|   +-- [filename].test.ts
+```
+
+Before you push your changes it's useful to run the test suite one more time. A PR that has failing tests will be regarded as a “Work in Progress” and will not be merged until all tests pass.
+
+```sh
+yarn test
+```
+
+If snapshot tests are broken, you can update your snapshots using `yarn test --update`. Storybook uses `jest` for testing, so you can use any of the jest flags that you're used to.
+
+## Submit a PR
+
+Finally, submit a PR against the `next` (default) branch.
+
+`next` is where all active development happens and corresponds to the latest prerelease of Storybook (e.g. `6.3.0-alpha.25`).
+
+If you are contributing a bugfix and you want it patched back to the `master` branch, which is where the latest stable version lives (e.g. `6.2.9`), mention that in your PR. We'll try to get it in if the fix looks non-disruptive and corresponds to a critical bug.
```

Comment:
```suggestion
If you are contributing a bugfix and you want it patched back to the `master` branch, which is where the latest stable version lives (e.g. `6.2.9`), mention that in your PR. We'll try to get it in if the fix looks non-disruptive and corresponds to a critical bug.

## Troubleshooting and feedback

Storybook has a [Discord community](https://discord.gg/c8meVWsWJg) and is open for everyone. There is a dedicated channel for contributors called `#contributing`, so make sure to check it out for feedback or support in the contribution process. The community is the easiest way to reach for maintainers and contributors!

```

Reply:
I think it's important to add a section for communication. What do you think?
I made sure the generated link has no invite limit and no expiration.

Reply:
Yes, that makes sense. The official link is https://discord.gg/storybook. Let's use that!